From 9548443ff96a248d9abc46ede4236046ec80824a Mon Sep 17 00:00:00 2001
From: Drive Player <drive-player@localhost>
Date: Mon, 28 Sep 2026 14:01:58 +0900
Subject: [PATCH] Build the bindings without the renderer discoverer

RendererDiscoverer.java is the one source of the bindings published under
the GNU General Public License; the rest is LGPL-2.1. JNI_OnLoad looks that
class up and fails the load without it, so the lookups and the JNI half are
dropped here, and the class itself is left out of the packaged sources.
Renderer discovery needs stream output and the renderer modules, which this
build does not include.
---
 libvlc/jni/Android.mk  |  2 +-
 libvlc/jni/libvlcjni.c | 20 --------------------
 libvlc/jni/utils.h     |  8 --------
 3 files changed, 1 insertion(+), 29 deletions(-)

diff --git a/libvlc/jni/Android.mk b/libvlc/jni/Android.mk
index bcabc48..e18b541 100644
--- a/libvlc/jni/Android.mk
+++ b/libvlc/jni/Android.mk
@@ -52,7 +52,7 @@ LOCAL_MODULE    := libvlcjni
 LOCAL_SRC_FILES := libvlcjni.c
 LOCAL_SRC_FILES += libvlcjni-mediaplayer.c
 LOCAL_SRC_FILES += libvlcjni-vlcobject.c
-LOCAL_SRC_FILES += libvlcjni-media.c libvlcjni-medialist.c libvlcjni-mediadiscoverer.c libvlcjni-rendererdiscoverer.c
+LOCAL_SRC_FILES += libvlcjni-media.c libvlcjni-medialist.c libvlcjni-mediadiscoverer.c
 LOCAL_SRC_FILES += libvlcjni-dialog.c
 LOCAL_SRC_FILES += std_logger.c
 LOCAL_C_INCLUDES := $(VLC_SRC_DIR)/include $(VLC_BUILD_DIR)/include
diff --git a/libvlc/jni/libvlcjni.c b/libvlc/jni/libvlcjni.c
index f7c74d3..c3654aa 100644
--- a/libvlc/jni/libvlcjni.c
+++ b/libvlc/jni/libvlcjni.c
@@ -189,10 +189,6 @@ jint JNI_OnLoad(JavaVM *vm, void *reserved)
               "org/videolan/libvlc/MediaDiscoverer", true);
     GET_CLASS(fields.MediaDiscoverer.Description.clazz,
               "org/videolan/libvlc/MediaDiscoverer$Description", true);
-    GET_CLASS(fields.RendererDiscoverer.clazz,
-              "org/videolan/libvlc/RendererDiscoverer", true);
-    GET_CLASS(fields.RendererDiscoverer.Description.clazz,
-              "org/videolan/libvlc/RendererDiscoverer$Description", true);
     GET_CLASS(fields.Dialog.clazz,
               "org/videolan/libvlc/Dialog", true);
 
@@ -278,20 +274,6 @@ jint JNI_OnLoad(JavaVM *vm, void *reserved)
            "(Ljava/lang/String;Ljava/lang/String;I)"
            "Lorg/videolan/libvlc/MediaDiscoverer$Description;");
 
-    GET_ID(GetStaticMethodID,
-           fields.RendererDiscoverer.createDescriptionFromNativeID,
-           fields.RendererDiscoverer.clazz,
-           "createDescriptionFromNative",
-           "(Ljava/lang/String;Ljava/lang/String;)"
-           "Lorg/videolan/libvlc/RendererDiscoverer$Description;");
-
-    GET_ID(GetStaticMethodID,
-           fields.RendererDiscoverer.createItemFromNativeID,
-           fields.RendererDiscoverer.clazz,
-           "createItemFromNative",
-           "(Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;IJ)"
-           "Lorg/videolan/libvlc/RendererItem;");
-
     GET_ID(GetStaticMethodID,
            fields.Dialog.displayErrorFromNativeID,
            fields.Dialog.clazz,
@@ -363,8 +345,6 @@ void JNI_OnUnload(JavaVM* vm, void* reserved)
     (*env)->DeleteGlobalRef(env, fields.MediaPlayer.Equalizer.clazz);
     (*env)->DeleteGlobalRef(env, fields.MediaDiscoverer.clazz);
     (*env)->DeleteGlobalRef(env, fields.MediaDiscoverer.Description.clazz);
-    (*env)->DeleteGlobalRef(env, fields.RendererDiscoverer.clazz);
-    (*env)->DeleteGlobalRef(env, fields.RendererDiscoverer.Description.clazz);
     (*env)->DeleteGlobalRef(env, fields.Dialog.clazz);
 
     pthread_key_delete(jni_env_key);
diff --git a/libvlc/jni/utils.h b/libvlc/jni/utils.h
index a03840c..8316791 100644
--- a/libvlc/jni/utils.h
+++ b/libvlc/jni/utils.h
@@ -98,14 +98,6 @@ struct fields {
         jclass clazz;
         jmethodID createDescriptionFromNativeID;
     } MediaDiscoverer;
-    struct {
-        struct {
-            jclass clazz;
-        } Description;
-        jclass clazz;
-        jmethodID createDescriptionFromNativeID;
-        jmethodID createItemFromNativeID;
-    } RendererDiscoverer;
     struct {
         jclass clazz;
         jmethodID displayErrorFromNativeID;
-- 
2.55.0

